src/main/java/com/marklogic/kafka/connect/source/RowBatcherSupplier.java

```diff
@@ -0,0 +1,73 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.datamovement.RowBatchFailureListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.row.RawPlanDefinition;
+import com.marklogic.client.row.RawQueryDSLPlan;
+import com.marklogic.client.row.RowManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+public abstract class RowBatcherSupplier implements Supplier<RowBatcher<?>> {
```

Comment:
Fun convenience class - extending LoggingObject (from ml-javaclient-util), which declares a logger for you and also a `format` method so you don't have to type `String.format`. Then you don't have to declare logger here or in its subclasses.